pkg/stanza/operator/parser/container/parser.go

```diff
@@ -73,20 +74,20 @@ func (p *Parser) Process(ctx context.Context, entry *entry.Entry) (err error) {
 	if format == "" {
 		format, err = p.detectFormat(entry)
 		if err != nil {
-			return fmt.Errorf("failed to detect a valid container log format: %w", err)
+			return p.HandleEntryError(ctx, entry, stanzaerr.Wrap(err, "invalid container log format"))
 		}
 	}
 
 	switch format {
 	case dockerFormat:
 		err = p.ParserOperator.ProcessWithCallback(ctx, entry, p.parseDocker, p.handleAttributeMappings)
```

Comment:
We can move additional logic from here to the callback function but then be unable to reuse this method everywhere else. It does not seem to add much optimization

Reply:
I believe when we use `ProcessWithCallback` method, it processes and writes to output before it has the chance to go though `recombineParser`. We also see this in test cases failing with `Received unexpected entry: ` message

Reply:
Interesting! Then that means the block at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/83aa8bf38f4ab3f6e3dacffb1b49ae8e4be3f7c2/pkg/stanza/operator/parser/container/parser.go#L82-L90 is also wrong since `parseTime` is called after?

In that case maybe it's more proper to introduce a new `ParseWithCallback` method fixing the above section as well?

Reply:
so I did add the `ParseWithCallback` and the tests are passing. Not entirely sure why the `docker` block parsed time correctly afterwards. One guess would be that the writer is asynchronous, and the `entry` value changed before it was written. Let me know what you think